MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m

```diff
@@ -446,6 +454,35 @@ - (nullable NSURL *)buildTokenRequestURLWithAppSecret:(NSString *)appSecret rele
   return components.URL;
 }
 
+- (void)openURLInAuthenticationSessionWith:(NSURL *)url fromClass:(Class)clazz {
+  MSLogDebug([MSDistribute logTag], @"Using SFAuthenticationSession to open URL: %@", url);
+  
+  NSString *callbackUrlScheme = [NSString stringWithFormat:kMSDefaultCustomSchemeFormat, self.appSecret];
+  
+  if (@available(iOS 11.0, *)) {
+    SFAuthenticationSession *session = [[clazz alloc] initWithURL:url callbackURLScheme:callbackUrlScheme completionHandler:^(NSURL *callbackUrl, NSError *error) {
+      
+      self.authenticationSession = nil;
+      MSLogDebug([MSDistribute logTag], @"Called %@ with errror: %@", callbackUrl, error.localizedDescription);
+      
+      if(error.code == SFAuthenticationErrorCanceledLogin) {
+        MSLogError([MSDistribute logTag], @"Authentication session was cancelled by user or failed.");
+      }
+      
+      if(callbackUrl) {
+        [self openURL:callbackUrl];
+      }
+    }];
+    self.authenticationSession = session;
+
+    BOOL success = [session start];
+    if(success) {
+      MSLogDebug([MSDistribute logTag], @"Authentication Session Started, showing confirmation dialog");
+    }
+  }
+}
+
+// TODO: rename method
```

Comment:
Let's rename now! What's your suggestion?

Reply:
`openURLInSafariViewController`??

Reply:
I've renamed it to`openURLInSafariViewControllerWithUrl:fromClass:` to make it work with the conventions...I ommited passing in the instance of SFSafariViewController though as we don't need that and it's an internal API.